Design review issues fix

0nko opened the pull request:
Fixes #1726. All mentioned issues should be fixed now. Also, a confirmation dialog was added to the refund flow.

![image](https://user-images.githubusercontent.com/1522856/71295871-0847dc00-237e-11ea-92ef-3bf5b4ee584d.png)

![image](https://user-images.githubusercontent.com/1522856/71295891-1a297f00-237e-11ea-8730-39cbf8b201c2.png)

![image](https://user-images.githubusercontent.com/1522856/71295934-388f7a80-237e-11ea-8f8a-954b379d6767.png)

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

nbradbury commented:
I have an order which I already refunded
![Screenshot_1576951734](https://user-images.githubusercontent.com/3903757/71311950-724e9880-23f3-11ea-8e10-ff223833a948.png)

nbradbury commented:
I refunded all the products for an order the tapped "Issue Refund" again and ended up seeing this screen:

![Screenshot_1576951734](https://user-images.githubusercontent.com/3903757/71311961-a6c25480-23f3-11ea-98f3-7f8c0c463857.png)

nbradbury commented:
In this refund screen, I selected to refund the product but the "Products refund" at the bottom of the screen shows $0. I believe it's because the product was already refunded, but if that's the case should "issue refund" be hidden?

Note that this was order #4422 in the "Bananza World Zhhhh" test store.


![Screenshot_1576952236](https://user-images.githubusercontent.com/3903757/71312052-aa0a1000-23f4-11ea-8a07-a45d717d317b.png)

0nko commented:
Thank you @kyleaparker and @nbradbury. The refund button and refund product count errors should be fixed now and the PR's ready for another round.

I've also updated the refund info and refund item layout based on the design updates:

![image](https://user-images.githubusercontent.com/1522856/73173908-c2153380-4106-11ea-8092-b3ae25e50f31.png)

One thing I wanted to mention - I've removed the feature flag in this PR before I left in December. Is it still OK or should I add it back?

nbradbury commented on WooCommerce/src/androidTest/kotlin/com/woocommerce/android/ui/products/MockedProductDetailViewModel.kt:
```diff
@@ -31,10 +30,10 @@ final class MockedProductDetailViewModel @AssistedInject constructor(
-            LiveDataDelegate(arg0, ViewState(), "", onChange = {
-                combineData(it.product!!, Parameters("$", "oz", "in"))
-            })
+//    override val viewStateData: LiveDataDelegate<ViewState> =
```
Should this be removed rather than commented out?

0nko replied:
This connected test that uses this class has been failing for a while, this was just so that the build works. We should address those eventually but probably not in this PR.

nbradbury replied:
Okay, perhaps add a `TODO` here then?

nbradbury commented:
@0nko This issue may be unrelated to this PR, or not even an issue at all, but I wanted to make sure.

I generated an order for three of the same product, which cost 0.10 each. I then refunded two of them, as shown below.

![refund](https://user-images.githubusercontent.com/3903757/73209923-348e0f80-4117-11ea-8345-ed64de37ecb9.png)

I then went to fulfill the order, and the total shows as the original unrefunded 0.30. Shouldn't this be showing 0.10 instead?

![fulfill](https://user-images.githubusercontent.com/3903757/73210083-86cf3080-4117-11ea-8ccb-20ca67740a52.png)

nbradbury commented:
This is more of a question for @kyleaparker, but the confirmation dialog states that the refund "can't be undone," then if you confirm it the next thing you see is a snackbar with an Undo button on it.

![dialog](https://user-images.githubusercontent.com/3903757/73211862-07dbf700-411b-11ea-815c-272ef45d0781.png)

![Screenshot_1580157484](https://user-images.githubusercontent.com/3903757/73211928-2510c580-411b-11ea-9f6b-b8313070ca00.png)

0nko commented:
@nbradbury You seem to be testing an older commit. The layout of the refund item in order detail has changed.

nbradbury commented:
> You seem to be testing an older commit. The layout of the refund item in order detail has changed.

Yeah, noticed that right after I posted :) But I have the latest changes and same problems exist.

kyleaparker commented:
>This is more of a question for @kyleaparker, but the confirmation dialog states that the refund "can't be undone," then if you confirm it the next thing you see is a snackbar with an Undo button on it.

I think we can remove the undo from the snackbar 👍

nbradbury commented:
The wording here for a manual refund is wonky. Should "via" be removed from the end here?

![via](https://user-images.githubusercontent.com/3903757/73212643-a452c900-411c-11ea-9df3-8252762e0c82.png)

nbradbury commented:
This may not be an issue since it sounds like we're removing the Undo button, but if I undo a refund I then get stuck in the refund fragment. I can try to back out, but it keeps me on the same screen and just shows a toast stating, "Refund in progress please wait."

![Screenshot_1580157752](https://user-images.githubusercontent.com/3903757/73212787-f85dad80-411c-11ea-84ca-65672020056a.png)

nbradbury commented:
I'm trying to figure out why there's no Issue Refund button here. This is the previously-mentioned order for three items, two of which were refunded. I don't see a way to refund the remaining item. I **can** refund it in web app.

![Screenshot_1580158391](https://user-images.githubusercontent.com/3903757/73212939-41adfd00-411d-11ea-84da-03d82f0268d2.png)

nbradbury commented:
@0nko Done with my first round!

0nko commented:
Thanks for the review, @nbradbury!

> I then went to fulfill the order, and the total shows as the original unrefunded 0.30. Shouldn't this be showing 0.10 instead?

Yes. This is fixed now.

> I think we can remove the undo from the snackbar 👍

Done.

> The wording here for a manual refund is wonky. Should "via" be removed from the end here?

This was caused by the empty payment gateway title, which is weird because I can see from other screenshots it's supposed to be COD. I added a logic that can handle it now, though.

> I'm trying to figure out why there's no Issue Refund button here. 

This was caused by a condition that only shows the button if there is a recorded payment from the customer. The web allows to refund items even if there is none so I've now removed it.

I also fixed the refund list reordering issue in order detail that @kyleaparker found.

Ready for another round 🤞 .

nbradbury commented:
You'll be happy to hear we can :shipit:

nbradbury commented:
@0nko Oh, wait - I see this is going to be merged into `release/3.5` which is already frozen. Should this be merged into `develop` instead? If it really should be in `release/3.5` feel free to merge it yourself since I've approved it.

0nko commented:
I've talked to @jkmassel about this PR and we're going to release a new beta after it is merged.
